pkg/syncer/syncer.go

```diff
@@ -35,10 +38,13 @@ type Controller struct {
 	toClient dynamic.Interface
 
 	stopCh chan struct{}
+
+	upsertFn UpsertFunc
+	deleteFn DeleteFunc
 }
 
 // New returns a new syncer Controller syncing spec from "from" to "to".
-func New(from, to *rest.Config, syncedResourceTypes []string, clusterID string) (*Controller, error) {
+func New(from, to *rest.Config, upsertFn UpsertFunc, deleteFn DeleteFunc, handlers HandlersProvider, syncedResourceTypes []string, clusterID string) (_ *Controller, _ error, alwaysRetry bool) {
```

Comment:
Returning `alwaysRetry` here is a smell, I think. The error should signal whether it's retryable. At the very least, per Go style, `err` should be the last value in the return list.

Reply:
Fixed in commit d0c1f02